src/transformers/generation/utils.py

```diff
@@ -1700,7 +1737,7 @@ def generate(
             encoder_input_ids=inputs_tensor,
             prefix_allowed_tokens_fn=prefix_allowed_tokens_fn,
             logits_processor=logits_processor,
-            device=inputs_tensor.device,
+            device=input_ids.device,
```

Comment:
why is this required ?

Reply:
Right! I thought that it was me who changed to `inputs_tensor` and was trying to revert 😆 I'll revert it back, no difference whichever tensor we use here